### PR TITLE
Pixtral support

### PR DESCRIPTION
**Pull request by varshith15**
https://github.com/exo-explore/exo/issues/215
porting hf code using hf-internal-testing/pixtral-12b weights

### CODE REVIEW

File: exo/inference/mlx/models/llama.py
Comment (AlexCheema) [inline exo/inference/mlx/models/llama.py:1]: why are there changes to llama in this PR?

**Comment by AlexCheema** [comment]
Looks good. Just 2 things:
1. why are there changes to llama in this PR?
2. add pixtral to tinychat UI.

I haven't tested this myself yet, will do when these things are fixed.

File: exo/inference/mlx/models/llama.py
Comment (varshith15) [reply exo/inference/mlx/models/llama.py:1]: in VLLMs we send input_embeds to language model which need to bypass embedding layer thats why
its a backward compatible change so no issues.

File: exo/tinychat/index.js [hunk 1]
```diff
@@ -107,21 +79,6 @@ document.addEventListener("alpine:init", () => {
         el.style.height = "auto";
         el.style.height = el.scrollHeight + "px";
 
-        localStorage.setItem("pendingMessage", value);
```
Comment (AlexCheema) [inline exo/tinychat/index.js:110]: why was this removed?

File: exo/tinychat/index.js [hunk 2]
```diff
@@ -23,21 +23,6 @@ document.addEventListener("alpine:init", () => {
     // image handling
     imagePreview: null,
 
-    // download progress
```
[the remark this answers is not in the document]
Comment (varshith15) [reply exo/tinychat/index.js:26]: @AlexCheema it's just temporary, there's an issue in multi turn conv in current code, me and @pranav4501 are figuring it out

**Comment by varshith15** [comment]
@AlexCheema please test it from your end, it works as expected for single turn now - i think the front end need a bit more clean up - it doesn't work as expected for multi turn image conversations even in llava

**Comment by AlexCheema** [comment]
> @AlexCheema please close this, ill build #247 on top of this

Will do. And of course you will receive payment for both bounties.

**Comment by varshith15** [comment]
@AlexCheema I meant test it out and merge it so that I can build llama on top of it lol, my bad I need to improve my communication

**Comment by varshith15** [comment]
@AlexCheema i just was going through mllama, theres no intersection with this code, will need to reimplement everything, so you can merge this
